src/utils/DockerInstallStatusProvider.ts

```diff
@@ -16,21 +16,28 @@ class DockerInstallStatusProvider implements IDockerInstallStatusProvider {
 
     public constructor() {
         this.isDockerInstalledLazy = new AsyncLazy<boolean>(async () => {
-            try {
-                await execAsync('docker -v');
+            const dockerInstalled = await this.isDockerInstalledRealTimeCheck();
+            if (dockerInstalled) {
                 // once docker is installed, lets assume that it will not be uninstalled.
                 // It is unlikely user will uninstall the docker, in which case user has open a new workspace to get the updated status.
                 this.isDockerInstalledLazy.clearLifeTime();
-                return true; // As long as the docker command did't throw exception, assume it is installed.
-            } catch (error) {
-                return false; // docker not installed
             }
+            return dockerInstalled;
         }, this.maxCacheTime);
     }
 
     public async isDockerInstalled(): Promise<boolean> {
         return await this.isDockerInstalledLazy.getValue();
     }
+
+    public async isDockerInstalledRealTimeCheck(): Promise<boolean> {
```

Comment:
How come we need a realtime check for the install command?

Reply:
We don't want to use cached value which may not be accurate especially for the case where it was installed and then uninstalled.

Reply:
Also we refresh every 30 seconds, so it could have a stale data. For this install flow, we should do real time check.

Reply:
That makes sense; user has it installed and launches VSCode; uninstalls, then uses the command to install.